packages/react-instantsearch-hooks/src/lib/useInstantSearchApi.ts

```diff
@@ -177,6 +180,7 @@ export function useInstantSearchApi<TUiState extends UiState, TRouteState>(
   const cleanupTimerRef = useRef<ReturnType<typeof setTimeout> | null>(null);
   const store = useSyncExternalStore<InstantSearch<TUiState, TRouteState>>(
     useCallback(() => {
+      // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
```

Comment:
weird, why was this needed now?

Reply:
Yeah I tried to align my TS version on the CI one but I still have the eslint error, I'm not sure why

Reply:
I guess as it doesn't happen on CI, let's leave it as-is :)

Reply:
Thanks!